@globberwops, Thanks for your pr, when I tested the usage I got this error, please take a look:

CMakeLists.txt:
````
# CMakeList.txt : CMake project for CMakeProject42, include source and define
# project specific logic here.
#
cmake_minimum_required (VERSION 3.8)

project(test)

find_package(tcp_pubsub CONFIG REQUIRED)

add_executable (main "CMakeProject42.cpp" "CMakeProject42.h")

target_link_libraries(main PRIVATE tcp_pubsub::tcp_pubsub)

````


CMakeProject42.cpp:
````
// CMakeProject42.cpp : Defines the entry point for the application.
//

#include "CMakeProject42.h"
#include "publisher.h"

using namespace std;

int main()
{
	cout << "Hello CMake." << endl;
	return 0;
}

````
CMakeProject42.h:
````
// CMakeProject42.h : Include file for standard system include files,
// or project specific include files.

#pragma once

#include <iostream>

// TODO: Reference additional headers your program requires here.

````

````
[CMake] CMake Error at F:/Feature-test/tcp/vcpkg/installed/x64-windows/share/tcp_pubsub/tcp_pubsubConfig.cmake:27 (INCLUDE):
1> [CMake]   INCLUDE could not find requested file:
1> [CMake] 
1> [CMake]     F:/Feature-test/tcp/vcpkg/installed/x64-windows/lib/cmake/tcp_pubsub/tcp_pubsubTargets.cmake
1> [CMake] Call Stack (most recent call first):
1> [CMake]   F:/Feature-test/tcp/vcpkg/scripts/buildsystems/vcpkg.cmake:843 (_find_package)
1> [CMake]   CMakeProject42/CMakeLists.txt:10 (find_package)
1> [CMake] -- Looking for pthread.h
1> [CMake] -- Looking for pthread.h - not found
1> [CMake] -- Found Threads: TRUE  
1> [CMake] -- Configuring incomplete, errors occurred!
1> [CMake] See also "C:/Users/test/source/repos/CMakeProject42/out/build/x64-Debug/CMakeFiles/CMakeOutput.log".
1> [CMake] See also "C:/Users/test/source/repos/CMakeProject42/out/build/x64-Debug/CMakeFiles/CMakeError.log".
1> 'C:\WINDOWS\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 16 2019" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\vzhli17\source\repos\CMakeProject42\out\install\x64-Debug" -DCMAKE_TOOLCHAIN_FILE=F:/Feature-test/tcp/vcpkg/scripts/buildsystems/vcpkg.cmake "C:\Users\vzhli17\source\repos\CMakeProject42" 2>&1"' execution failed with error: ''C:\WINDOWS\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 16 2019" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\test\source\repos\CMakeProject42\out\install\x64-Debug" -DCMAKE_TOOLCHAIN_FILE=F:/Feature-test/tcp/vcpkg/scripts/buildsystems/vcpkg.cmake "C:\Users\test\source\repos\CMakeProject42" 2>&1"' returned with exit code: 1'.

````

--- a/ports/tcp-pubsub/fix-package-config-file.patch
+++ b/ports/tcp-pubsub/fix-package-config-file.patch
@@ -1,0 +1,13 @@
+diff --git a/tcp_pubsub/cmake/tcp_pubsubConfig.cmake.in b/tcp_pubsub/cmake/tcp_pubsubConfig.cmake.in
+index e4dd17a..553b0b1 100644
+--- a/tcp_pubsub/cmake/tcp_pubsubConfig.cmake.in
++++ b/tcp_pubsub/cmake/tcp_pubsubConfig.cmake.in
+@@ -1,4 +1,6 @@
+ @PACKAGE_INIT@
+ 
+-INCLUDE("@PACKAGE_TCP_PUBSUB_INSTALL_CMAKE_DIR@/tcp_pubsubTargets.cmake")
+-find_package(Threads)
++include(CMakeFindDependencyMacro)
++find_dependency(Threads)
++
++include("${CMAKE_CURRENT_LIST_DIR}/tcp_pubsubTargets.cmake")

--- a/ports/tcp-pubsub/portfile.cmake
+++ b/ports/tcp-pubsub/portfile.cmake
@@ -5,7 +5,9 @@ vcpkg_from_github(
     REPO eclipse-ecal/tcp_pubsub
     REF "v${VERSION}"
     SHA512 f89b9d9cdbd8e5787ac8923ec65cc2fc259e7d12269b1466a4c29657d8d466e39b95ec8b8483e975bf393f71b5c2d8f59cfd3d955e4e72d69716ec59fe0429af
-    PATCHES "use-ports-for-asio-and-recycle.patch"
+    PATCHES
+        "fix-package-config-file.patch"
+        "use-ports-for-asio-and-recycle.patch"
 )
 
 vcpkg_cmake_configure(

--- a/versions/t-/tcp-pubsub.json
+++ b/versions/t-/tcp-pubsub.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "a9660c8f7831ca125ca541c2af926c55c8425e50",
+      "git-tree": "075f0da2c8e03d545d28b625ecadde2a58135382",
       "version": "1.0.3",
       "port-version": 0
     }

